rest/audit_test.go

```diff
@@ -46,58 +52,163 @@ func TestAuditInjectableHeader(t *testing.T) {
 
 	// initialize RestTester
 	RequireStatus(t, rt.CreateDatabase("db", rt.NewDbConfig()), http.StatusCreated)
+	rt.CreateUser(requestUser, nil)
+
+	// auditFieldValueIgnored is a special value for an audit field to skip value-specific checks whilst still ensuring the field property is set
+	// used for unpredictable values, for example timestamps or request IDs (when the test is making concurrent or unordered requests)
+	const auditFieldValueIgnored = "sg_test_audit_field_value_ignored"
 
-	rt.CreateUser("alice", nil)
 	testCases := []struct {
 		name string
-		fn   func(method, resource, body string, headers map[string]string) *TestResponse
+		// auditableAction is a function that performs an action that should've been audited
+		auditableAction func(t testing.TB)
+		// expectedAuditEvents is a list of expected audit events and their fields for the given action... can be more than one event produced for a given action
+		expectedAuditEventFields map[base.AuditID]base.AuditFields
 	}{
-		/* FIXME: admin requests are not currently audited
 		{
-			name: "admin request",
-			fn:   rt.SendAdminRequestWithHeaders,
+			name: "guest request",
+			auditableAction: func(t testing.TB) {
+				RequireStatus(t, rt.SendRequest(http.MethodGet, "/db/_session", ""), http.StatusOK)
+			},
+			expectedAuditEventFields: map[base.AuditID]base.AuditFields{
+				base.AuditIDPublicUserAuthenticated: {
+					"cid":         auditFieldValueIgnored,
```

Comment:
I've been trending toward using constants for this and making the field names public from the base package.